### PR TITLE
Add data-test-id attributes to myaccount components

### PR DESCRIPTION
**Pull request by wytm97**
Please note $subject

fixes https://github.com/wso2/product-is/issues/9638

### CODE REVIEW

File: apps/myaccount/src/components/applications/all-applications.tsx [hunk 1]
```diff
@@ -16,14 +16,16 @@
...
  */
 
+import { TestableComponentInterface } from '@wso2is/core/models';
 import React, { FunctionComponent } from "react";
 import { ApplicationList } from "./application-list";
 import { Application } from "../../models";
 
 /**
- * Proptypes for the all applications component.
+ * Proptypes for the all applications component. Also see
```
Comment (jeradrutnam) [inline apps/myaccount/src/components/applications/all-applications.tsx:25]:
```suggestion
 * Proptypes for the all applications component.
```

File: apps/myaccount/src/components/account-recovery/options/security-questions-recovery.tsx [hunk 2]
```diff
@@ -47,9 +48,10 @@ import { EditSection, ThemeIcon } from "../../shared";
 const QUESTION = "question-";
 
 /**
- * Prop types for SecurityQuestionsComponent
+ * Prop types for SecurityQuestionsComponent. Also see
+ * {@link SecurityQuestionsComponent.defaultProps}
```
Comment (jeradrutnam) [inline apps/myaccount/src/components/account-recovery/options/security-questions-recovery.tsx:52]:
```suggestion
 * Prop types for SecurityQuestionsComponent.
 * Also see {@link SecurityQuestionsComponent.defaultProps}
```

File: apps/myaccount/src/components/applications/all-applications.tsx [hunk 3]
```diff
@@ -16,14 +16,16 @@
...
+ * {@link AllApplications.defaultProps}
```
Comment (jeradrutnam) [inline apps/myaccount/src/components/applications/all-applications.tsx:26]:
```suggestion
 * Proptypes for the all applications component.
 * Also see {@link AllApplications.defaultProps}
```

File: apps/myaccount/src/components/applications/recent-application-card.tsx [hunk 4]
```diff
@@ -16,16 +16,18 @@
  * under the License.
  */
 
+import { TestableComponentInterface } from '@wso2is/core/models';
 import classNames from "classnames";
 import React, { FunctionComponent } from "react";
 import { Card, Icon } from "semantic-ui-react";
 import { Application } from "../../models";
 import { AppAvatar } from "../shared";
+import {RecentApplications} from "./recent-applications";
```
Comment (jeradrutnam) [inline apps/myaccount/src/components/applications/recent-application-card.tsx:25]:
```suggestion
import { RecentApplications } from "./recent-applications";
```

File: apps/myaccount/src/components/applications/recent-application-card.tsx [hunk 5]
```diff
@@ -16,16 +16,18 @@
  * under the License.
  */
 
+import { TestableComponentInterface } from '@wso2is/core/models';
```
Comment (jeradrutnam) [inline apps/myaccount/src/components/applications/recent-application-card.tsx:19]:
```suggestion
import { TestableComponentInterface } from "@wso2is/core/models";
```

File: apps/myaccount/src/components/account-recovery/options/email-recovery.tsx [hunk 6]
```diff
@@ -15,6 +15,7 @@
  * specific language governing permissions and limitations
  * under the License.
  */
+import { TestableComponentInterface } from '@wso2is/core/models';
```
Comment (jeradrutnam) [inline apps/myaccount/src/components/account-recovery/options/email-recovery.tsx:18]:
```suggestion
import { TestableComponentInterface } from "@wso2is/core/models";
```

File: apps/myaccount/src/components/account-recovery/options/email-recovery.tsx [hunk 7]
```diff
@@ -394,3 +397,11 @@ export const EmailRecovery: React.FunctionComponent<EmailRecoveryProps> = (props
 
     return showEditView();
 };
+
+/**
+ * Default props of {@link EmailRecovery} component.
+ * Also see {@link EmailRecoveryProps}
+ */
+EmailRecovery.defaultProps = {
+    "data-testid": "email-recovery",
```
Comment (jeradrutnam) [inline apps/myaccount/src/components/account-recovery/options/email-recovery.tsx:406]:
```suggestion
    "data-testid": "email-recovery"
```

File: apps/myaccount/src/components/account-recovery/options/security-questions-recovery.tsx [hunk 8]
```diff
@@ -16,6 +16,7 @@
  * under the License.
  */
 
+import { TestableComponentInterface } from '@wso2is/core/models';
```
Comment (jeradrutnam) [inline apps/myaccount/src/components/account-recovery/options/security-questions-recovery.tsx:19]:
```suggestion
import { TestableComponentInterface } from "@wso2is/core/models";
```

File: apps/myaccount/src/components/account-recovery/options/security-questions-recovery.tsx [hunk 9]
```diff
@@ -516,3 +518,11 @@ export const SecurityQuestionsComponent: React.FunctionComponent<SecurityQuestio
 
     return <>{ listItems() }</>;
 };
+
+/**
+ * Default props for {@link SecurityQuestionsComponent}.
+ * Also see {@link SecurityQuestionsProps}
+ */
+SecurityQuestionsComponent.defaultProps = {
+  "data-testid": "security-questions-component",
```
Comment (jeradrutnam) [inline apps/myaccount/src/components/account-recovery/options/security-questions-recovery.tsx:527]:
```suggestion
  "data-testid": "security-questions-component"
```

File: apps/myaccount/src/components/applications/all-applications.tsx [hunk 10]
```diff
@@ -58,7 +70,9 @@ export const AllApplications: FunctionComponent<AllApplicationsProps> = (
 
 /**
  * All applications component default props.
+ * Also see {@link AllApplicationsProps}
  */
 AllApplications.defaultProps = {
-    showFavourites: true
+    showFavourites: true,
+    "data-testid": 'all-applications',
```
Comment (jeradrutnam) [inline apps/myaccount/src/components/applications/all-applications.tsx:77]:
```suggestion
    "data-testid": 'all-applications'
```

File: apps/myaccount/src/components/applications/all-applications.tsx [hunk 11]
```diff
@@ -16,14 +16,16 @@
  * under the License.
  */
 
+import { TestableComponentInterface } from '@wso2is/core/models';
```
Comment (jeradrutnam) [inline apps/myaccount/src/components/applications/all-applications.tsx:19]:
```suggestion
import { TestableComponentInterface } from "@wso2is/core/models";
```

File: apps/myaccount/src/components/account-recovery/account-recovery.tsx [hunk 12]
```diff
@@ -45,12 +45,13 @@ export const AccountRecoveryComponent: React.FunctionComponent<AccountRecoveryPr
     props: AccountRecoveryProps
 ): JSX.Element => {
     const { t } = useTranslation();
-    const { onAlertFired, featureConfig } = props;
+    const { onAlertFired, featureConfig, ['data-testid']: testId } = props;
```
Comment (brionmario) [inline apps/myaccount/src/components/account-recovery/account-recovery.tsx:48]: Shall we reformat this as follows. Please check other places as well.

```suggestion
    const {
        onAlertFired,
        featureConfig,
        ['data-testid']: testId
    } = props;
```

**Review by brionmario — APPROVED** [review]

File: apps/myaccount/src/components/account-recovery/account-recovery.tsx (hunk 12 above)
Comment (wytm97) [reply apps/myaccount/src/components/account-recovery/account-recovery.tsx]: Reformatted as required. However, we can keep object de-structuring statements inline if it's has only like 2 properties right? if property count is 3 or more let's reformat to multiple lines.

**Review by DimalChandrasiri — APPROVED** [review]

**Review by brionmario — APPROVED** [review]

**Pull request merged**